### PR TITLE
Add aot_arm_compiler flag to allow the reordering of the inputs

### PR DESCRIPTION
**Pull request by SaoirseARM**
### Summary

Adding a argument to aot_arm_compiler to allow for the ordering of the inputs.

* Add capability to use cmd input order in the backend
* Extend the test infrastructure to handle this

### CODE REVIEW

**Review by freddan80 — APPROVED** [review]

File: backends/arm/arm_vela.py [hunk 1]
```diff
@@ -17,10 +17,13 @@
...
 # Pack either input or output tensor block, compose the related arrays into
 # per-io structs to simplify runtime use.
-def vela_bin_pack_io(prefix, data):
-    ios = struct.pack("<i", len(data[prefix + "_shape"]))
-    for i in range(len(data[prefix + "_shape"])):
-        io_shape = data[prefix + "_shape"][i]
+def vela_bin_pack_io(prefix, data, shape_order=None):
+    vela_input_shapes = data[prefix + "_shape"]
+
+    order = shape_order if shape_order else range(len(vela_input_shapes))
```
[the remark this answers is not in the document]
Comment (SaoirseARM) [reply backends/arm/arm_vela.py:23]: Thank you for the review, and I agree. 
I will fix this in another commit. 

Thanks, 
Saoirse

**Pull request merged**